docs/tutorials/sentiment-analysis-of-movie-reviews.mdx

```diff
@@ -0,0 +1,200 @@
+# Sentiment Analysis on Movie Reviews using OpenAI and MongoDB
+
+## Introduction
+
+Good Day Devs! In this tutorial, we are going to analyse the sentiment on movie reviews using OpenAi and MongoDB. we'll use Movie Reviews Dataset from [`Hugging Face`](https://huggingface.co/datasets/rotten_tomatoes).
+
+- type of sentiment we are going to analyse for movie review
+  - positive
+  - neutral
+  - negative
+
+## Data Setup
+
+- Download [`MongoDB Compass`](https://www.mongodb.com/try/download/compass) or [`MongoDB Shell`](https://www.mongodb.com/try/download/shell)
+- Download the [`Movie reviews`](https://huggingface.co/datasets/rotten_tomatoes) dataset
+- Run the `MongoDB Compass` or `MongoDB Shell`
+
+## Connecting the Database
+
+- Open the MongoDb Compass Or Shell.
+- Connect with URI : `mongodb://cloud.mindsdb.com`.
+- If you are using the MongoDB Shell then shell prompt will ask for `login-email` and `password` then hit `Enter` key to connect.
+- If you are using the MongoDB Compass then,
+  - add `URI` in the URI input area
+  - In `Advanced Connection Options` select `Authentication`
+  - Then select `Username/Password` ,enter credentials in it and hit `Connect` button.
+
+After connecting with the MindsDB Cloud, let's bring our database using `insertOne` statement,
+
+- `host` : connection string for the mongodb which you can get from `MongoDB Atlas`.
+
+```bash
+test> use mindsdb
+mindsdb> db.databases.insertOne({
+            'name': 'mongo_demo_db',
+            'engine': 'mongodb',
+            'connection_args': {
+                "host": "mongodb+srv://<username>:<password>@<cluster host/ID>.mongodb.net/",
+                "database": "public"
+            }
+        })
+```
+
+Here is the output:
+
+```bash
+{
+  acknowledged: true,
+  insertedId: ObjectId("6415a242e60077d8d3548786")
+}
+```
+
+// By following these steps you're good to go with further steps i.e, analyse the movie review sentiment.
+
+## Understanding the Data
+
+We use the movie reviews dataset, where each row has a movie review and we'll predication whether the review is positive, negative or neutral.
+
+Let’s look at the data stored in the `movie_reviews` table that contains the input columns for the model.
+
+```bash
+mindsdb> use mongo_demo_db
+mongo_demo_db> db.movie_reviews.find({}).limit(3)
+```
+
+Here is the output:
+
+```bash
+{
+  _id : "64129b66e24d840208b2e498",
+  review : "it was good"
+},
+{
+  _id : "64129d44e24d840208b2e49a",
+  review : "this is a film well worth seeing , talking and singing heads and all ."
+},
+{
+  _id : "64129d64e24d840208b2e49b",
+  review : "ultimately , it ponders the reasons we need stories so much ."
+}
+```
+
+## Training a Predictor
+
+We use the `openai` engine to predict the sentiment of the review.
+
+we can create a model for text sentiment analysis like this:
+
+```bash
+mongo_demo_db> db.models.insertOne({
```

Comment:
This should be executed from mindsdb:

`mindsdb> db.models.insertOne({`

Please see this example: https://docs.mindsdb.com/nlp/sentiment-analysis-inside-mongodb-with-openai

Based on the example, please correct it throughout this article. The `insertOne`, `db.getCollection('models')`, etc. should be executed from `mindsdb>` and not from `mongo_demo_db>`.